File: Nov2020/23Nov/IntervalListIntersections.java
/*
Problem Name: 986. Interval List Intersections
Problem Link: https://leetcode.com/problems/interval-list-intersections/
*/
class Solution {
    public int[][] intervalIntersection(int[][] A, int[][] B) {
        int a = 0, b = 0;
        List<int[]> l = new ArrayList<int[]>();
        for(;a<A.length && b<B.length;){
            if(A[a][1] < B[b][0])
                a++;
            else if(A[a][0] > B[b][1])
                b++;
            else{
                l.add(new int[]{Math.max(A[a][0],B[b][0]), Math.min(A[a][1],B[b][1])});
                if(A[a][1] < B[b][1])
                    a++;
                else if(A[a][1] > B[b][1])
                    b++;
                else{
                    a++;
                    b++;
                }
            }
        }
        int[][] ans = new int[l.size()][];
        for(int i=0;i<l.size();i++){
            ans[i] = l.get(i);
        }
        return ans;
    }
}

File: Nov2020/23Nov/IntervalListIntersections.cpp
/*
Problem name: 986. Interval List Intersections
Problem link: https://leetcode.com/problems/interval-list-intersections/
*/

class Solution {
public:
    vector<vector<int>> intervalIntersection(vector<vector<int>>& A, vector<vector<int>>& B) {
        vector<vector<int>> common;
        int i=0,j=0,x,y;
        
        while( i<A.size() && j<B.size() ){
            x = max(A[i][0], B[j][0]);
            y = min(A[i][1], B[j][1]);
            
            if(x <= y)
                common.push_back({x,y});
            
            if(y == A[i][1]) i++;
            if(y == B[j][1]) j++;
        }
        return common;
    }
};
